rest-service/manager_rest/test/endpoints/test_nodes.py

```diff
@@ -205,178 +231,157 @@ def test_patch_node_runtime_props_overwrite(self):
+        self._instance('12', node=node1)
+        self._instance('21', node=node2)
+        self._instance('22', node=node2)
+        self._instance('31', node=node3)
+        self._instance('32', node=node3)
+        self._instance('41', node=node4)
+        self._instance('44', node=node4)
 
         all_instances = self.client.node_instances.list()
         dep1_instances = self.client.node_instances.list(
-            deployment_id='111')
+            deployment_id=self.dep1.id)
         dep2_instances = self.client.node_instances.list(
-            deployment_id='222')
+            deployment_id=dep2.id)
         dep1_n1_instances = self.client.node_instances.list(
-            deployment_id='111',
-            node_name='1'
+            deployment_id=self.dep1.id,
+            node_id='1'
         )
         dep1_n2_instances = self.client.node_instances.list(
-            deployment_id='111',
-            node_name='2'
+            deployment_id=self.dep1.id,
+            node_id='2'
         )
         dep2_n3_instances = self.client.node_instances.list(
-            deployment_id='222',
-            node_name='3'
+            deployment_id=dep2.id,
+            node_id='3'
         )
         dep2_n4_instances = self.client.node_instances.list(
-            deployment_id='222',
-            node_name='4'
+            deployment_id=dep2.id,
+            node_id='4'
         )
 
-        self.assertEqual(8, len(all_instances))
+        assert len(all_instances) == 8
+        assert len(dep1_instances) == 4
+        assert len(dep2_instances) == 4
+        assert all(ni.deployment_id == self.dep1.id for ni in dep1_instances)
+        assert all(ni.deployment_id == dep2.id for ni in dep2_instances)
 
-        def assert_dep(expected_len, dep, instances):
-            self.assertEqual(expected_len, len(instances))
-            for instance in instances:
-                self.assertEqual(instance.deployment_id, dep)
+        assert len(dep1_n1_instances) == 2
+        assert all(ni.deployment_id == self.dep1.id
+                   for ni in dep1_n1_instances)
+        assert all(ni.node_id == node1.id for ni in dep1_n1_instances)
 
-        assert_dep(4, '111', dep1_instances)
-        assert_dep(4, '222', dep2_instances)
+        assert len(dep1_n2_instances) == 2
+        assert all(ni.deployment_id == self.dep1.id
+                   for ni in dep1_n2_instances)
+        assert all(ni.node_id == node2.id for ni in dep1_n2_instances)
 
-        def assert_dep_and_node(expected_len, dep, node_id, instances):
-            self.assertEqual(expected_len, len(instances))
-            for instance in instances:
-                self.assertEqual(instance.deployment_id, dep)
-                self.assertEqual(instance.node_id, node_id)
+        assert len(dep2_n3_instances) == 2
+        assert all(ni.deployment_id == dep2.id for ni in dep2_n3_instances)
+        assert all(ni.node_id == node3.id for ni in dep2_n3_instances)
```

Comment:
those all written out seem silly, but I prefer this as it is simpler than a nested function